bridges/snowbridge/pallets/system-v2/src/lib.rs

```diff
@@ -221,6 +242,39 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		#[pallet::call_index(5)]
+		#[pallet::weight(<T as pallet::Config>::WeightInfo::add_tip())]
+		pub fn add_tip(
+			origin: OriginFor<T>,
+			sender: AccountIdOf<T>,
+			message_id: MessageId,
+			amount: u128,
+		) -> DispatchResult {
+			T::FrontendOrigin::ensure_origin(origin)?;
+
+			let result = match message_id {
+				Inbound(nonce) => <T as pallet::Config>::InboundQueue::add_tip(nonce, amount),
+				Outbound(nonce) => <T as pallet::Config>::OutboundQueue::add_tip(nonce, amount),
+			};
+
+			Self::deposit_event(Event::<T>::TipAdded {
```

Comment:
Can we emit the event as the last statement in the function? that seems to be general style across substrate (and ethereum)

Reply:
Resolved in https://github.com/vgeddes/polkadot-sdk/commit/a0fda22f8e4a362f46b101909969750fd2cd6ac7